### PR TITLE
Dev/ahn nath: styling and adding the backend to the project

### PR DESCRIPTION
**Pull request by ahn-nath**
- Styled sections and improved the UI of some pages.
- Create the UI and base template for the contributions page. 
- Added the backend.

### CODE REVIEW

File: frontend/src/pages/login.js [hunk 1]
```diff
@@ -1,177 +1,14 @@
-import React, { useState, useEffect, useContext } from "react";
```
Comment (tushar5526) [inline frontend/src/pages/login.js:1]: Why are we deleting the OAuth functionality?

**Comment by tushar5526** [comment]
Did you pull the changes from the main, switched to a new branch and then worked further?

File: frontend/src/pages/login.js (hunk 1 above)
Comment (ahn-nath) [reply frontend/src/pages/login.js:1]: Because we only need to use a personal token for the GitHub API. At least that is the easiest option,

**Comment by ahn-nath** [comment]
> Did you pull the changes from the main, switched to a new branch and then worked further?

Yes, but after I made the changes to my branch. I initially forgot to merge it before continuing on my branch and had to resolve conflicts. Let me know if I deleted something relevant.

**Comment by tushar5526** [comment]
Yeah sorry I forgot that we dont need OAuth. Its good to go!

**Pull request merged**